### PR TITLE
Fix PutObject call for unknown size stream

### LINKED ISSUE #169: PutObject call needs to support unknown sized stream
SDK needs to catch up on unknown sized stream support.

Needs functional test - see #156 
 3. Missing test for putobject() for unknown sized stream (single/multipart) (may be missing feature)

### PR DESCRIPTION
**Pull request by poornas**
Fixes #169 

Currently PutObject call with unknown stream size loops infinitely. Fixed that bug and added a functional test.

### CODE REVIEW

**Comment by poornas** [comment]
@donatello and @krishnasrinivas , pinging for reviews.

File: Minio/ApiEndpoints/ObjectOperations.cs [hunk 1]
```diff
@@ -248,11 +248,17 @@ public partial class MinioClient : IObjectOperations
...
             int partNumber;
+            int numPartsUploaded = 0;
             bool skipUpload = false;
             for (partNumber = 1; partNumber <= partCount; partNumber++)
             {
                 byte[] dataToCopy = ReadFull(data, (int)partSize);
-
+                if (dataToCopy == null)
+                {
+                    skipUpload = true;
```
Comment (donatello) [inline Minio/ApiEndpoints/ObjectOperations.cs]: Looks like skipUpload is not read outside the for loop, so this line can be removed.

Comment (poornas) [reply Minio/ApiEndpoints/ObjectOperations.cs]: good catch, removed that line.

**Comment by poornas** [comment]
@krishnasrinivas , this is waiting on your review.

**Pull request merged**